sklearn/neighbors/classification.py

```diff
@@ -374,17 +391,51 @@ def predict(self, X):
             else:
                 mode = np.array([weighted_mode(pl, w)[0]
                                  for (pl, w)
-                                 in zip(pred_labels[inliers], weights[inliers])],
```

Comment:
The code here and above had been deleted in #8096: https://github.com/scikit-learn/scikit-learn/pull/8096/files#diff-364ed6a16c035d2ac7a77e55f1ab7e3bL367

Why did you reinsert it?